### PR TITLE
Use operator/ in fs::absolute to prepare for C++17

### PR DESCRIPTION
**Pull request by kiminuo**
This PR contains only two commits cherry-picked from #19245 as suggested [here](https://github.com/bitcoin/bitcoin/pull/19245#issuecomment-642735541). Motivations for the changes are in the commit messages.

Cheers!

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

**Comment by MarcoFalke** [comment]
In the second commit: Why not change those as well: https://github.com/bitcoin/bitcoin/pull/19245/commits/6cf2c8eae6b8c2ea503c9b03cdc5a90ad31d14bd#diff-431ad0fd7e189837a5fcebddb320315e ?

Also, could add commit 6f55de61d8249658de7ee8563fdefc2fafd0d932 or would that be a behaviour change on non-posix machines?

**Comment by kiminuo** [comment]
> In the second commit: Why not change those as well: [6cf2c8e#diff-431ad0fd7e189837a5fcebddb320315e](https://github.com/bitcoin/bitcoin/commit/6cf2c8eae6b8c2ea503c9b03cdc5a90ad31d14bd#diff-431ad0fd7e189837a5fcebddb320315e) ?

I have added that.

> Also, could add commit [6f55de6](https://github.com/bitcoin/bitcoin/commit/6f55de61d8249658de7ee8563fdefc2fafd0d932) or would that be a behaviour change on non-posix machines?

I don't know whether it can break something. When switching system_complete() to absolute() in C++17, it is OK according to https://stackoverflow.com/a/46271698/13716157 so when we wait, it won't break. That's the reason I haven't added the commit.

**Comment by MarcoFalke** [comment]
According to https://github.com/boostorg/filesystem/commit/7e300b986bb978db54305effceacd94cfc42e6cf it is identical to `absolute` (except for the kinky Windows behaviour, which I hope we don't rely on)

**Comment by kiminuo** [comment]
> According to [boostorg/filesystem@7e300b9](https://github.com/boostorg/filesystem/commit/7e300b986bb978db54305effceacd94cfc42e6cf) it is identical to `absolute` (except for the kinky Windows behaviour, which I hope we don't rely on)

Ok, will add that.

@MarcoFalke `wallet_multiwallet.py` test fails on line 101: https://github.com/bitcoin/bitcoin/blob/abdfd2d0e3ebec7dbead89317ee9192189a35809/test/functional/wallet_multiwallet.py#L101:

```py
 assert_equal(sorted(map(lambda w: w['name'], self.nodes[0].listwalletdir()['wallets'])), ['', os.path.join('sub', 'w5'), 'w', 'w1', 'w2', 'w3', 'w7', 'w7_symlink', 'w8'])
```

And I wonder why the path corresponding to `w6` is missing on the right? What is the reason for that?

Locally I can see in my test output:

```
AssertionError: not(['', 'sub/w5', 'tmp/test_runner_₿_🏃_20200708_160230/wallet_multiwallet_0/extern/w6', 'w', 'w1', 'w2', 'w3', 'w7', 'w7_symlink', 'w8'] == ['', 'sub/w5', 'w', 'w1', 'w2', 'w3', 'w7', 'w7_symlink', 'w8'])
```

so that the left side of the assert contains that `w6` path but the right side does not. Right side is the *expected* result, right?

btw: `util.py` / `def assert_equal(thing1, thing2, *args):` can be confusing because it does not state whether `thing1` is expected result or actual result. Most unit testing frameworks use that notation.

**Comment by MarcoFalke** [comment]
w6 is extern, so it should *not* show up in the wallet dir

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -2288,10 +2288,10 @@ UniValue dumptxoutset(const JSONRPCRequest& request)
         }
     }.Check(request);
 
-    fs::path path = fs::absolute(request.params[0].get_str(), GetDataDir());
+    fs::path path = fs::absolute(GetDataDir() / request.params[0].get_str());
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:2291]: Ah my bad. This is a change in behavior.

Joining "/tmp/" and "/foo" will correctly give:

* `/foo` with the code in current master
* `/foo` with std lib absolute (your version)
* `/tmp//sadf` with the current version

**Review by maflcko — CHANGES_REQUESTED** [review]

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (kiminuo) [reply src/rpc/blockchain.cpp:2291]: Yeah.

Even this https://github.com/bitcoin/bitcoin/pull/19466/files#diff-e802a36c28b0140bab62cb5366199656R97 breaks multiwallet test.

So the PR without those changes is essentially empty and it looks to me I can close it.

Comment (MarcoFalke) [reply src/rpc/blockchain.cpp:2291]: > So the PR without those changes is essentially empty and it looks to me I can close it.

Jup, was my fault for suggesting to split them up

**Pull request closed without merging**